### DISCUSSION TITLE
Handling error with max-retry uses dead letter policies

### DISCUSSION THREAD

**Discussion by ahmedsaalah** [gocardless/hutch#383] [General]
I need to handle errors in case there is an error or exception to retry with number of retries with decrement number of reminder retries if still failed pass to dead letter exchanges/queue 
First Question: Hutch Support to make dead letter exchange ?
Second Question: There is another way to implement this solution instead of read message and duplicate this message with new retries reminder reject the original and publish the new one -as I feel it's so noob solution- ? as there is only two methods i can use in the consumer requeue and reject!

**Comment by michaelklishin** [comment]
Hutch does not support anything like that, it's very minimalistic. [Sneakers does](https://github.com/jondot/sneakers/).

The only other option is to use the delayed message exchange (I doubt this would be necessary), which
badly needs a rework due to its limitations. However, the earliest that can ship would be around RabbitMQ 3.12.0 in 2023.

**Reply by ahmedsaalah** [reply]
Ok, Thanks.
Just to make sure in Hutch there is no way to make this from hutch instead of bunny
```
x = ch.direct('work.exchange')
dlx = ch.direct('retry.exchange')

target_queue = ch.queue('work.queue', arguments: {
  :'x-dead-letter-exchange' => 'retry.exchange'
}).bind(x, routing_key: 'work.queue')

retry_queue = ch.queue('retry.queue', arguments: {
  :'x-dead-letter-exchange' => 'work.exchange',
  :'x-message-ttl' => 60000
}).bind(dlx, routing_key: 'work.queue')
```

**Reply by michaelklishin** [reply]
You can use dead lettering, of course, but then you would be rolling your own feature
